homeassistant/components/totalconnect/diagnostics.py

```diff
@@ -0,0 +1,113 @@
+"""Provides diagnostics for TotalConnect."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.diagnostics import async_redact_data
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+TO_REDACT = [
+    "username",
+    "Password",
+    "Usercode",
+    "UserID",
+    "Serial Number",
+    "serial_number",
+    "sensor_serial_number",
+]
+
+# Private variable access needed for diagnostics
+# pylint: disable=protected-access
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    client = hass.data[DOMAIN][config_entry.entry_id].client
+
+    data = {}
+    data["client"] = {
+        "auto_bypass_low_battery": client.auto_bypass_low_battery,
+        "module_flags": client._module_flags,
+        "retry_delay": client.retry_delay,
+        "invalid_credentials": client._invalid_credentials,
+    }
+
+    data["user"] = {
+        "master": client._user._master_user,
+        "user_admin": client._user._user_admin,
+        "config_admin": client._user._config_admin,
+        "security_problem": client._user.security_problem(),
+        "features": client._user._features,
+    }
+
+    data["locations"] = {}
+    for location_id, location in client.locations.items():
+        new_location = {
+            "location_id": location.location_id,
```

Comment:
Is including this necessary?
Won't it be displayed as the higher level key anyway?

Reply:
Instead of removing this line, I removed the key and made it a list instead of a dict.  I think it will be more easier/more obvious for troubleshooting to keep "location_id" next to the number instead of a key all by itself.